doc/coding.md

```diff
@@ -0,0 +1,333 @@
+```javascript
+router.get('/:name/about', function(req, res) {
+
+    return fs.readFileAsync(__dirname + '/../static/'
+            + encodeURIComponent(req.params.name) + '.html')
+    .then(function(src) {
+        res.status(200).type('html').send(src)
+    }).catch(function(err) {
+        throw new HTTPError({
+            status: 404,
+            type: 'not_found',
+            title: 'Not Found',
+            detail: 'No information could be found on ' + req.params.name
+        });
+    });
+
+});
+```
+
+Note that you can also attach additional debug information to the `HTTPError`
+object to help you track down bugs. This information is going to be logged, but
+will not reach the client, thus ensuring no sensitive information is leaked
+unintentionally. To do so, simply add any property you deem important when
+creating / throwing the error.
+
+## Logging and Metrics
+
+Logging and metrics collection is supported out of the box via
+[service-runner](https://github.com/wikimedia/service-runner). They are exposed
+in route handler files via the `app.logger` and `app.metrics` objects.
+
+### Logging
+
+To log something, simply use `app.logger.log(level, what)`. The logger itself is
+a [bunyan](https://github.com/trentm/node-bunyan) wrapper, and thus supports the
+following levels:
+
+- `trace`
+- `debug`
+- `info`
+- `warn`
+- `error`
+- `fatal`
+
+Additionally, it is good practice to attach a component name to the log level as
+it eases log indexing and filtering later in production. For example, if a log
+entry has the `debug` level and pertains to one of our example routes, the log
+level could be set to `debug/people`. The `what` portion of the log entry can be
+either a string message, or any *stringifiable* object. As an example, let's
+log the person's name given to the `/people/:name/about` route and the file name
+that is going to be looked up:
+
+```javascript
+router.get('/:name/about', function(req, res) {
+
+    var info = {
+        name: req.params.name,
+        path: __dirname + '/../static/'
+            + encodeURIComponent(req.params.name) + '.html'
+    };
+
+    app.logger.log('debug/people/about', info);
+
+    return fs.readFileAsync(info.path)
+    .then(function(src) {
+        res.status(200).type('html').send(src)
+    }).catch(function(err) {
+        throw new HTTPError({
+            status: 404,
+            type: 'not_found',
+            title: 'Not Found',
+            detail: 'No information could be found on ' + info.name
+        });
+    });
+
+});
+```
+
+### Metrics Collection
+
+Collecting metrics is a great way to have insights into the overall health and
+performance of your service. When using the template, this is as easy as calling
+one of the following methods:
+
+- `app.metrics.timing`
+- `app.metrics.increment`
+- `app.metrics.decrement`
+- `app.metrics.histogram`
+- `app.metrics.gauge`
+- `app.metrics.unique`
+
+How can one them? Let's show it on `/people/:name/news`. This route uses an
```

Comment:
Yeas, sometimes I think faster than I type and forget to write some words :smile: